### ISSUE TITLE
"Add device" capability

### ISSUE THREAD

**Issue by iver56** [iver56/homey-ismartgate#4]
With open/close controls and visualization of current state

**Comment by filipbech** [comment]
Would you like help with this?

**Comment by iver56** [comment]
Yes, that would be nice

**Comment by filipbech** [comment]
Can you point me to the documentation to the api please? Then I’m happy to give it a try and submit a PR. If you have a preference for how you’d like it done, please let me know

**Comment by iver56** [comment]
I am afraid I am not aware of any API documentation. I ported the code from an open source ismartgate client written in Python. Is there any other way I can assist?

**Comment by filipbech** [comment]
I guess my main issue is if we can be notified on open/close so we don't have to poll for state. (Since it can also be opened from other sources). 

I tried to make a virtual device and create a flow that did the polling and I suspect it hade some side-effects (the gate closed randomly). Haven't debugged enough to see if I did something wrong or where that came from. 

What was the python project you were migrating?

Anyway, I'll give it a go.

**Comment by iver56** [comment]
As far as I know, the device does not actively push updates on open/close state. So to achieve such a feature, the homey app would have to poll regularly.

> I suspect it hade some side-effects (the gate closed randomly)

That's curious! I cannot immediately understand why that would happen just by polling the current state

> What was the python project you were migrating?

If I remember correctly, it was this one: https://github.com/vangorra/python_gogogate2_api

**Comment by filipbech** [comment]
I'll take a look maybe over christmas, and see what we can come up with!

**Comment by iver56** [comment]
Sounds good to me. Good luck!

**Comment by iver56** [comment]
@filipbech Any progress thus far?

By the way, I have asked in the homey community how to do the transition without breaking the existing setup people have. The answer can be found here: https://community.homey.app/t/how-to-gracefully-introduce-breaking-change-in-existing-homey-app/101957

**Comment by filipbech** [comment]
No, sorry... nothing to report yet. Its still high on my want-to-build list, but time's not easy to find apparently
